File: github_hooks/db/migrate/20260807120000_add_build_status_guards_table.rb
class AddBuildStatusGuardsTable < ActiveRecord::Migration[8.0]
  def change
    create_table :build_status_guards,
                 primary_key: [:repository_id, :commit_sha, :context, :source_id] do |t|
      t.uuid :repository_id, null: false
      t.text :commit_sha, null: false
      t.text :context, null: false
      t.text :source_id, null: false
      t.string :last_state, null: true
      t.timestamptz :claimed_at, null: true
      t.timestamptz :updated_at, null: false

      t.index :updated_at, name: "build_status_guards_updated_at_index"
    end
  end
end

File: repository_hub/priv/repo/migrations/20260807120000_add_build_status_guards_table.exs
defmodule RepositoryHub.Repo.Migrations.AddBuildStatusGuardsTable do
  @moduledoc false
  use Ecto.Migration

  def change do
    create table(:build_status_guards, primary_key: false) do
      add(:repository_id, :uuid, null: false, primary_key: true)
      add(:commit_sha, :text, null: false, primary_key: true)
      add(:context, :text, null: false, primary_key: true)
      add(:source_id, :text, null: false, primary_key: true)
      add(:last_state, :string, null: true)
      add(:claimed_at, :timestamptz, null: true)
      add(:updated_at, :timestamptz, null: false)
    end

    create(index(:build_status_guards, [:updated_at]))
  end
end
